src/character.cpp

```diff
@@ -2142,10 +2161,24 @@ item &Character::i_add( item it, bool should_stack )
             break;
         }
     }
-    auto &item_in_inv = inv.add_item( it, keep_invlet, true, should_stack );
-    item_in_inv.on_pickup( *this );
-    cached_info.erase( "reloadables" );
-    return item_in_inv;
+    item_pocket *pocket = best_pocket( it );
+    if( pocket == nullptr ) {
+        if( !has_weapon() ) {
+            weapon = it;
+            return weapon;
+        }
+        debugmsg( "no space to add item. dropping" );
```

Comment:
this is the lack of api i mentioned in the body of the PR summary. i want some opinions on a general way to handle this, maybe a change in what this function returns or something? some way it can fail properly.

Reply:
i've got what i think is a reasonable solution in place "for now" but i'm thinking this still could use something reasonable that's not an item reference. perhaps an item location.